src/hooks/usePost.js

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from 'react';
+import endpoints from './endpoints.js';
+
+const usePost = (newEmailSubscriber) => {
+  const [message, setMessage] = useState(null);
+  const [isSubmitting, setIsSubmitting] = useState(true);
+  const [error, setError] = useState(null);
+  const { first_name, last_name, state, email, language } = newEmailSubscriber;
+
+  useEffect(() => {
+    const postNewUser = async () => {
+      const response = await fetch(endpoints.submit, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+          first_name: first_name,
+          last_name: last_name,
+          state: state,
+          email: email,
+          language: language
+        })
+
+      if (response.status !== 200) {
+        throw new Error (`${response.status} : Something has gone wrong, please try again.`)
+      }
+      if (!response.ok) {
+        throw new Error ('Something has gone wrong, please try again.')
+      }
+
+      const message = await response.json();
+
+      return message;
+    }
+
+    postNewUser()
+      .then(message => {
+        setMessage(message);
+        setIsSubmitting(false);
+        setError(null);
+      })
+      .catch(error => {
+        setError(error.message);
+        setIsSubmitting(false);
+      })
+  }, [newEmailSubscriber])
+
+  return { message, isSubmitting, error };
+}
+
+export default usePost;
```

Comment:
These hooks files were removed in the previous `eb-api-post` PR and it looks like you've successfully refactored by adding this functionality directly to the form. What do you think about removing `src/hook/useFetch.js` and  `src/hook/usePost.js`from this PR?